homeassistant/components/cloud/http_api.py

```diff
@@ -389,6 +391,56 @@ async def post(self, request: web.Request, data: dict[str, Any]) -> web.Response
         return self.json_message("ok")
 
 
+class DownloadSupportPackageView(HomeAssistantView):
+    """Download support package view."""
+
+    url = "/api/cloud/support_package"
+    name = "api:cloud:support_package"
+
+    async def get(self, request: web.Request) -> web.Response:
+        """Download support package file."""
+
+        def get_domain_table_markdown(domain_info: dict[str, Any]) -> str:
+            if len(domain_info) == 0:
+                return "No information available\n"
+
+            markdown = ""
+            first = True
+            for key, value in domain_info.items():
+                markdown += f"{key} | {value}\n"
+                if first:
+                    markdown += "--- | ---\n"
+                    first = False
+            return markdown + "\n"
+
+        hass = request.app[KEY_HASS]
+        domain_health = await get_system_health_info(hass)
+        markdown = "## System Information\n\n"
+
+        try:
+            hass_info = domain_health.pop("homeassistant")
```

Comment:
I'd consider moving the whole markdown generation from system health to a separate function, not nested.